correct CompanyController.java

DominikSerwatka opened the pull request:
- improve CompanyController.java and add GlobalExceptionHandler.java

DominikSerwatka committed:
correct CompanyController.java

DominikSerwatka committed:
add hibernate.reveng.xml to exclude table

DominikSerwatka committed:
resolve conflicts

DominikSerwatka commented:
- add hibernate.revenge.xml file to exclude "flyway_schema_history" table
- there is still a problem with auto increment of company id,  
due to the lack of "@GeneratedValue(strategy = GenerationType.IDENTITY)", auto incrementation does not work

jdabrowski commented:
Let's not rely on autoincrement from database. Id should match UUIDv4 (https://en.wikipedia.org/wiki/Universally_unique_identifier) format if provided. If not provided it should be generated by us

jdabrowski reviewed (changes requested):
in comments

DominikSerwatka committed:
add GlobalExceptionHandler class and change CompanyController class and change hibernate.reveng.xml to use UUID

DominikSerwatka committed:
add GlobalExceptionHandler class and change CompanyController class and change hibernate.reveng.xml to use UUID

DominikSerwatka commented:
- Deleted the WelcomeController class
- Modified the hibernate.reveng.xml file to generate UUID as the ID type instead of Serializable
- Updated the POST request handling so:  Now, it generates a new UUID for the ID and If a client sends an ID in the request, the server responds with a message: "id should not be provided" - I'm not sure if it is ok
